File: aisile-manager-web/src/main/webapp/js/service/itemCatService.js
//封装所有连接后端代码    需要$http
app.service('itemCatService',function($http){
	//查询全部的数据
	this.findAll = function(){
		return $http.get('../itemCat/findAll.do');
	}
	//分页
	this.findPage = function(page,rows){
		return $http.get('../itemCat/findPage.do?page='+page+"&rows="+rows);
	}
	//分页带数据查询
	this.findSearch = function(page,rows,searchEntity){
		return $http.post('../itemCat/findSearch.do?page='+page+"&rows="+rows,searchEntity);
	}
	//修改
	this.update = function(entity){
		return $http.post('../itemCat/update.do',entity);
	}
	//添加
	this.add = function(entity){
		return $http.post('../itemCat/add.do',entity);
	}
	//根据id查询对象
	this.findOne = function(id){
		return $http.post('../itemCat/findOne.do?id='+id);
	}
	//删除
	this.dele = function(selectIds){
		return $http.get('../itemCat/dele.do?ids='+selectIds);
	}
	//根据父亲的id查询列表
	this.findAllByParentId = function(id){
		return $http.get('../itemCat/findAllByParentId.do?parentId='+id);
	}
})

File: aisile-sellergoods-interface/src/main/java/com/aisile/sellergoods/service/ItemCatService.java
package com.aisile.sellergoods.service;

import java.util.List;

import com.aisile.pojo.TbItemCat;
import com.aisile.pojo.entity.PageResult;

public interface ItemCatService {
	/**
	 * 查询全部数据
	 * @return
	 */
	public List<TbItemCat> findAll();
	/**
	 * 分页
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	/**
	 * 模糊查询
	 * @param pageNum
	 * @param pageSize
	 * @param TbItemCat
	 * @return
	 */
	public PageResult findSearch(int pageNum,int pageSize,TbItemCat tbItemCat);
	/**
	 * 添加方法
	 * @param parentId 
	 * @param tnBrand
	 */
	public void add(TbItemCat tbItemCat);
	/**
	 * 修改方法
	 * @param TbItemCat
	 */
	public void update(TbItemCat tbItemCat);
	/**
	 * 回显   返回的是一个对象
	 * @param id
	 * @return
	 */
	public TbItemCat findOne(Long id);
	/**
	 * 删除(批量删除)
	 * @param ids
	 */
	public boolean delete(Long id);
	/**
	 * 根据父亲的id获取列表
	 * @param parentId
	 * @return
	 */
	public List<TbItemCat> findAllByParentId(Long parentId);
	public List<TbItemCat> findOne1(Long id);
}
